src/components/ScreensharingPanel.tsx

```diff
@@ -0,0 +1,62 @@
+import { DeviceInfo } from "../containers/StreamingSettingsCard";
+import { useState } from "react";
+import { showToastError } from "./Toasts";
+import { TbScreenShare } from "react-icons/tb";
+
+type ScreensharingPanelProps = {
+  addLocalStream: (stream: MediaStream, id: string) => void;
+  label: string;
+  setSelectedDeviceId: (trackId: DeviceInfo | null) => void;
+};
+
+const SCREEENSHARING_VIDEO_CONSTRAINTS = {
+  frameRate: { ideal: 20, max: 25 },
+  width: { max: 1920, ideal: 1920 },
+  height: { max: 1080, ideal: 1080 },
+};
+
+export const ScreensharingPanel = ({ label, addLocalStream, setSelectedDeviceId }: ScreensharingPanelProps) => {
+  const [screenshareAudio, setScreenshareAudio] = useState<boolean>(false);
+  return (
+    <div className="card-body p-1 flex bg-base-100 shadow-xl m-2 w-full flex-row rounded-md flex-1 items-center ">
+      <button
+        className="btn btn-success btn-sm m-2"
+        onClick={() => {
+          const videoId = "screenshare_" + crypto.randomUUID();
+          const audioId = "screenshare_" + crypto.randomUUID();
+          if (navigator.mediaDevices.getDisplayMedia) {
```

Comment:
I think we should introduce some kind of warning if context is not secure 
```
if (!window.isSecureContext) {
  // display warning
}
```

We could link that warning to this if statement because the error "Screensharing is not supported in this browser" might not be accurate. The browser does support screen sharing, but the context may not be secure